File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        node = TreeNode(val)
        if not root:
            return node
        if not root.left and root.val > val:
            root.left = node
        elif not root.right and root.val < val:
            root.right = node
        if root.val > val:
            self.insertIntoBST(root.left, val)
        elif root.val < val:
            self.insertIntoBST(root.right, val)
        return root

File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function(root, val) {
    const node = new TreeNode(val)
    if (!root) return node
    if (!root.left && root.val > val) {
        root.left = node
    } else if (!root.right && root.val < val) {
        root.right = node
    }
    if (root.val > val) {
        insertIntoBST(root.left, val)
    } else if (root.val < val) {
        insertIntoBST(root.right, val)
    }
    return root
};